### PR TITLE
Fixed bug on overflown number literals in scanner

### LINKED ISSUE #534: tooooooooooo loooooooongggggggggggggggg id/version number over flow
what if the id 
insert vertex xxx 100000000000000000000000000000000000000000()
the result is over flow:

nebula> go from 10100000000000000000000000000000000000000000000000000000 over like                                                                    =======================                                                                                                                               |                  id |                                                                                                                               =======================                                                                                                                               | 9223372036854775807 |                                                                                                                               -----------------------                                                                                                                               Got 1 rows (Time spent: 9218/9974 us)          

I don't think over flow is a good idea.

### PR DESCRIPTION
**Pull request by dutor**
As title.

This PR will close #534

### CODE REVIEW

File: src/parser/scanner.lex [hunk 1]
```diff
@@ -249,20 +249,64 @@ IP_OCTET                    ([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])
...
+                                    while (i < yyleng && yytext[i] == '0') {
+                                        i++;
+                                    }
+                                    if (yyleng - i > 16) {
+                                        yyterminate();
+                                    }
+                                }
                                 int64_t val = 0;
                                 sscanf(yytext, "%lx", &val);
                                 yylval->intval = val;
```
Comment (steppenwolfyuetong) [inline src/parser/scanner.lex]: The maximum value of int64 is 9,223,372,036,854,775,807,
Hexadecimal is 0X7FFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF
when value is in [ 0X8000 0000 0000 0000 0000 0000 0000 0000, 0XFFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF] ， the values still overflow

File: src/parser/scanner.lex [hunk 2]
```diff
@@ -249,20 +249,64 @@ IP_OCTET                    ([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])
...
                             }
 0[Xx]{HEX}+                 {
+                                if (yyleng > 18) {
+                                    auto i = 2;
+                                    while (i < yyleng && yytext[i] == '0') {
+                                        i++;
+                                    }
+                                    if (yyleng - i > 16) {
+                                        yyterminate();
+                                    }
```
Comment (steppenwolfyuetong) [inline src/parser/scanner.lex]: The maximum value of int64 is 9,223,372,036,854,775,807,
Hexadecimal is 0X7FFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF
when value is in [ 0X8000 0000 0000 0000 0000 0000 0000 0000, 0XFFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF] ， the value is still overflow.

So, here add esle statement

File: src/parser/scanner.lex [hunk 3]
```diff
@@ -249,20 +249,64 @@ IP_OCTET                    ([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])
...
+                                }
                                 int64_t val = 0;
                                 sscanf(yytext, "%lx", &val);
```
Comment (steppenwolfyuetong) [inline src/parser/scanner.lex]: Should we judge the return value of sscanf?

File: src/parser/scanner.lex (hunk 2 above)
Comment (dutor) [reply src/parser/scanner.lex]: Hex is a way to specify integers as raw bytes, there is no such thing overflow in terms of literals.

File: src/parser/scanner.lex (hunk 3 above)
Comment (dutor) [reply src/parser/scanner.lex]: Yes, we should have done that, even though the regex and checking above already guarantee the correctness.

Comment (steppenwolfyuetong) [reply src/parser/scanner.lex]: 👍

File: src/parser/scanner.lex (hunk 2 above)
Comment (steppenwolfyuetong) [reply src/parser/scanner.lex]: if （yyleng - i == 16 && yytext[i] > 7 ),   Does the hexadecimal  value not overflow?
the value in decimal is great than 9,223,372,036,854,775,807.

File: src/parser/test/ScannerTest.cpp [hunk 4]
```diff
@@ -350,6 +373,17 @@ TEST(Scanner, Basic) {
         CHECK_SEMANTIC_VALUE(".123", TokenType::DOUBLE, 0.123),
         CHECK_SEMANTIC_VALUE("123.456", TokenType::DOUBLE, 123.456),
 
+        CHECK_SEMANTIC_VALUE("0xFFFFFFFFFFFFFFFF", TokenType::INTEGER, 0xFFFFFFFFFFFFFFFFL),
```
Comment (steppenwolfyuetong) [inline src/parser/test/ScannerTest.cpp]: when use int64, the value "0xFFFFFFFFFFFFFFFF"  has already has overflowed.

File: src/parser/scanner.lex (hunk 2 above)
Comment (steppenwolfyuetong) [reply src/parser/scanner.lex]: Yeah， you are right,   [-9,223,372,036,854,775,808     9,223,372,036,854,775,807] is [0, 0XFFFF FFFF FFFF FFFF]
👍

Comment (monadbobo) [reply src/parser/scanner.lex]: you can replace  magic number "18" with  std::numeric_limits<int64_t>::digits10 .

Comment (dutor) [reply src/parser/scanner.lex]: @monadbobo But here we are dealing with hexdecimal, instead of base 10.

**Pull request merged**